doc/sphinx-guides/source/api/native-api.rst

```diff
@@ -3686,6 +3686,34 @@ To update the blocks that are linked, send an array with those blocks.
 
 To remove all links to blocks, send an empty array.
 
+Set Available Licenses for a Dataset Type
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+With this API a superuser may limit the available licenses for a dataset of a given type to a subset of those available in the installation.
+
+For example, a superuser could create a type called "software" and limit the available licenses to only "CC0 1.0"
+
+This API endpoint is for superusers only.
+
+.. code-block:: bash
+
+  export API_TOKEN=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+  export SERVER_URL=https://demo.dataverse.org
+  export TYPE=software
+  export JSON='["CC0 1.0"]'
+
+  curl -H "X-Dataverse-key:$API_TOKEN" -H "Content-Type: application/json" "$SERVER_URL/api/datasets/datasetTypes/$TYPE/licenses" -X PUT -d $JSON
+
+The fully expanded example above (without environment variables) looks like this:
+
+.. code-block:: bash
+
+  curl -H "X-Dataverse-key:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx" -H "Content-Type: application/json" "https://demo.dataverse.org/api/datasets/datasetTypes/software/licenses" -X PUT -d '["CC0 1.0"]'
+
+To update the licenses available, send an array with those licenses.
+
+To remove all links to licenses, send an empty array. If no licenses are set then the default is all active licenses in the installation.
```

Comment:
Do we have the use case where someone would want to require custom terms by not allowing any licenses for a given type? (i.e. no content would not mean delete?)

Reply:
I don't know. I can ask the curation team.

Reply:
@sbarbosadataverse, @jggautier - do we have any cases where we want to require the Dataset owner to put in Custom Terms?

Reply:
I'll also have to fix/update "getAvailableLicenses" and the doc, etc.

Reply:
FWIW: I was thinking that instead of having no licenses mean delete that you actually allow that value and have a separate delete. Then the default would be a new type gets all the normal licenses unless you override per type. That would avoid duplicating db entries (and having types get out of sync if you add a new license, etc.)

Reply:
OK. that makes sense. keeping up to date with available licenses would be a problem.

Reply:
OK, so bottom line is that if we say that available  licenses for a type are depicted in the datasettype_license table then of a type has no records there we can assume that all active licenses are available for datasets of that type.

Reply:
Yes, except I'm not sure that supports a way to say that a type should only allow custom terms/no licenses -  unless that table can have null entries (or there's some other boolean) to indicate that you shouldn't inherit all the active ones. Again, not sure if the design has to cover that.

Reply:
What's the status here?

Are we adding a separate DELETE?

Are custom terms supported?

Reply:
My take from standup is that we'll go with what's implemented so far and follow up with another pull request if we need different functionality.